images/generate_cache.py

```diff
@@ -0,0 +1,187 @@
+"""
+ @file
+ @brief This file generates the image cache for openshot-qt, essentially creating multiple
+ resolution versions of each image used in the UI, for faster loading and high DPI support.
+ @author Jonathan Thomas <jonathan@openshot.org>
+
+ @section LICENSE
+
+ Copyright (c) 2008-2018 OpenShot Studios, LLC
+ (http://www.openshotstudios.com). This file is part of
+ OpenShot Video Editor (http://www.openshot.org), an open-source project
+ dedicated to delivering high quality video editing and animation solutions
+ to the world.
+
+ OpenShot Video Editor is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OpenShot Video Editor is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with OpenShot Library.  If not, see <http://www.gnu.org/licenses/>.
+ """
+
+import json
+import os
+from classes import info
+from classes.logger import log
+import openshot
+from PyQt5.QtWidgets import QApplication
+
+# Try to get the security-patched XML functions from defusedxml
+try:
+    from defusedxml import minidom as xml
+except ImportError:
+    from xml.dom import minidom as xml
+
+
+# Create Qt application (required for SVG font rendering)
+svg_app = QApplication([])
+icon_size = info.LIST_ICON_SIZE
+emoji_size = info.EMOJI_ICON_SIZE
+
+
+def create_cache_thumbnail(path, width, height, thumb_path):
```

Comment:
Sugestion: I'd model the concept of image and thumbnail on classes, read them as a list and then use `create_cache_thumbnail(image: Image)` instead of passing all the parameters that compose the image. Doing so as (if) the the number of parameters around image/thumnail grows you do not need to have the number of parameters in the function signature to grow:

The classes signature would be the following:

```
@dataclasses.dataclass(frozen=True)
class Image:
    path: pathlib.Path
    width: int
    height: int
    thumbnail: Thumbnail

@dataclasses.dataclass(frozen=True)
class Thumbnail:
    path: pathlib.Path
```

Also that makes a clear interface for (if) other modules want to use the exposed method.